content/React Query and management of server state.md

```diff
@@ -0,0 +1,75 @@
+In the image above, we can see the different states for the server state information, let's see what is the meaning of each state:
+- Fetching: This is the initial state and occurs when the data is fetching from an outsource, typically the backend.
+- Fresh: On this state, the information is the same in both sides, this is the desired state of our application because implies that we don't need to re-fetch the information. On the other hand, this state lasts a short time because in most cases when your fetching information, this information immediately is potentially out of date.
+- Stale: Represents the data that's out of date and is used on the app currently, this is because of frontend has old information and need to re-fetch from backend, or backend is not updated because frontend didn't send the new information to backend yet. This state is particularly important as long as we want to be in the fresh state again.
+- Inactive: React Query has a garbage collector for managing cache on the browser, this state in some manner indicates on React query that if the data is not be using on the application then it's gonna be potentially deleted after some time. This is a great feature of React Query, because we want to persist the recent information for not fetching data all the time and improve the speed/UX of our interfaces.
+- Deleted: This happens when the data was inactive for a determinate time, in which case is deleted from the cache. This time can be configurable locally for each query or globally.
+
+With this approach, React Query handles the asynchronous information of our application clearly, allow us to scale and maintain our code in a better way.
+
+## Introduction to React query API
+Although React Query has only two hooks, it is highly configurable in all aspects, since the retry delay time of the queries until what is the interval time of re-fetch in background. 
+But let's start with the most basic API that is well documented in [their repository](https://github.com/tannerlinsley/react-query)
+``` javascript 
+const { status, data, error } =  useQuery('movies', fetchTodoList);
+```
+This hook provided us the status of the fetch(loading, error, or success), data and error if are defined. Up to this point is a pretty normal hook for fetching information, but as we see the query is related to a key(in this case `movies`), this is a unique-global key that is used for associate the query information between the components, in this way we can re-use this hook use the same information in anywhere of the DOM tree. If you follow the classic redux action-reducer cycle for fetching data, **this hooks it will save up a lot of repetitive code**.
+
+Among other things, React Query also allows us to filter the information with a simple system.
+```javascript
+// An individual movie
+useQuery(['movies', 5], ...);
+
+// A list of movies that are "released"
+useQuery(['movies', { type:  'released' }], ...);
+```
```

Comment:
We are usually using react gists for code samples.

Reply:
I think react gists is useful when you have a mini demo or something more functional that you would like to be able to copy and test locally as a reader, but in this case, it is just an API demo.
By the other hand, it's important to maintain consistency between the different posts of the blog, so I will change it, thanks for the suggestion 😄